pkg/syncer/spec/spec_controller.go

```diff
@@ -193,14 +192,14 @@ func NewSpecSyncer(syncerLogger logr.Logger, syncTargetWorkspace logicalcluster.
 
 	secretMutator := specmutators.NewSecretMutator()
 
-	upstreamSecretIndexer := upstreamInformers.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}).Informer().GetIndexer()
-	deploymentMutator := specmutators.NewDeploymentMutator(upstreamURL, newSecretLister(upstreamSecretIndexer), syncTargetWorkspace, dnsIP)
+	// make sure the secrets informer gets started
+	_ = upstreamInformers.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}).Informer()
+	deploymentMutator := specmutators.NewDeploymentMutator(upstreamURL, func(clusterName logicalcluster.Name, namespace string) ([]runtime.Object, error) {
+		items, err := upstreamInformers.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}).Lister().ByCluster(clusterName).ByNamespace(namespace).List(labels.Everything())
+		fmt.Printf("SKUZNETS: %s|%s got %d items, (%v)\n", clusterName, namespace, items, err)
```

Comment:
Love it 😛

Reply:
:facepalm:

Reply:
I name-prefix them to make it easy to find & remove ... but this one was all-caps for some reason.